backend/src/worker/tasks.py

```diff
@@ -1,6 +1,43 @@
+from io import BytesIO
+
+from fastapi import HTTPException
+from src.api_parsers.dblp_parser import DBLPParser
+from src.api_parsers.exceptions import NoAuthorsException
+from src.api_parsers.scopus_parser import ScopusParser
+from src.articles_service.articles_parser import ArticleParser, KeywordParsingError
+from src.common.postgres import get_db_session
+from src.models.author import Author
+from src.search.repositories import AuthorRepository
 from src.worker.core import celery
 
 
-@celery.task(name="hello.world", bind=True)
-def hello_world(self, name):
-    print("Hello world")
+@celery.task(name="search", bind=True)
+def search(self, file_contents: bytes, search_id: int) -> None:
+    try:
+        db_session = get_db_session()
+        article_parser = ArticleParser(BytesIO(file_contents))
+        keywords = article_parser.get_keywords()
+        authors = []
+
+        for keyword in keywords[0]:
+            parser = ScopusParser(keywords=keyword.replace("\n", " "))
+            try:
+                scopus_authors = parser.get_authors()
+            except NoAuthorsException:
+                continue
+            authors.extend(scopus_authors)
+
+        for keyword in keywords[1]:
+            parser = DBLPParser(keywords=keyword.replace("\n", " "))
+            try:
+                dblp_authors = parser.get_authors()
+            except NoAuthorsException:
+                continue
+            authors.extend(dblp_authors)
+
+        authors = [
+            Author(search_id=search_id, **author.model_dump()) for author in authors
+        ]
+        AuthorRepository.create_all(db_session, authors)
+    except KeywordParsingError:
+        raise HTTPException(500, detail="An error occurred whilst parsing the article.")
```

Comment:
I dont know about raising http exception here, its not an endpoint

Reply:
Ok, I'll move it to the endpoint.

Reply:
What im trying to say is that we tell celery hey put in the queue this task and we tell user that this task was queued. We can catch queueing exceptions and then raise http 500 exception but here it does not make sense

Reply:
![image](https://github.com/pwr-TeamProject23/M2/assets/78159695/b4c3a5f0-8a0c-41e0-80bb-575a0baf3e1e)
These are excpetions raisen by apply_async

Reply:
+ the external api services also could raise some exception, why not catch them all, log the error and some information and then raise it again for celery to save error status?